File: app/src/main/java/com/kelsos/mbrc/mappers/ConnectionMapper.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.data.ConnectionSettings;
import com.kelsos.mbrc.data.DiscoveryMessage;

public class ConnectionMapper implements Mapper<DiscoveryMessage, ConnectionSettings> {
  @Override
  public ConnectionSettings map(DiscoveryMessage discoveryMessage) {
    ConnectionSettings settings = new ConnectionSettings();
    settings.setAddress(discoveryMessage.getAddress());
    settings.setPort(discoveryMessage.getPort());
    settings.setName(discoveryMessage.getName());
    return settings;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/ConnectionMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.data.ConnectionSettings
import com.kelsos.mbrc.data.DiscoveryMessage

class ConnectionMapper : Mapper<DiscoveryMessage, ConnectionSettings> {
  override fun map(from: DiscoveryMessage): ConnectionSettings {
    return ConnectionSettings().apply {
      address = from.address
      port = from.port
      name = from.name
    }
  }
}
